### ISSUE TITLE
Add answered_a, answered_b and answered(part) to Puzzle

### ISSUE THREAD

**Issue by lukehankins** [wimglenn/advent-of-code-data#34]
I've found these useful:

```python
    @property
    def answered_a(self):
        return bool(getattr(self, "answer_a", None))

    @property
    def answered_b(self):
        return bool(getattr(self, "answer_b", None))

    def answered(self, part):
        if part == "a":
            return bool(getattr(self, "answer_a", None))
        if part == "b":
            return bool(getattr(self, "answer_b", None))
        raise AttributeError
```

Thanks for the helpful tool!

(I'd submit these as a pull request but I've run out of cycles getting pytest to run clean; I'm missing a plugin and I can't figure out which.)

**Comment by wimglenn** [comment]
All the test requirements are listed in https://github.com/wimglenn/advent-of-code-data/blob/master/tests/requirements.txt so you can `pip install -r tests/requirements.txt`.

**Comment by lukehankins** [comment]
Unfortunately, I'm running into https://github.com/mariocesar/pytest-requests/issues/1.  I'll keep poking at it. (after I do day7, of course)

**Comment by wimglenn** [comment]
`pytest-requests` is not in our dependencies...

**Comment by lukehankins** [comment]
Looks like I was reusing a virtualenv.  Sloppy.  :-/  PR submitted.

**Comment by wimglenn** [comment]
I had been using `hasattr(puzzle, "answer_a")` instead for this but I guess your way is more obvious.  Thanks for the contribution, it is released to PyPI as v0.9.3.
